File: Klanvision_V5_Backend-master/controllers/applicationController.js
import { Application } from '../models/Application.js';

export const submitApplication = async (req, res) => {
  try {
    const { jobTitle, name, dob, email, phone, gender, qualification, experience, skills, linkedin, portfolio } = req.body;
    
    let resumeData = null;
    let resumeFileName = null;
    let resumeContentType = null;
    
    if (req.file) {
      resumeData = req.file.buffer;
      resumeFileName = req.file.originalname;
      resumeContentType = req.file.mimetype;
    }
    
    await Application.create({
      jobTitle,
      name,
      dob,
      email,
      phone,
      gender,
      qualification,
      experience,
      skills,
      linkedin,
      portfolio,
      resumeData,
      resumeFileName,
      resumeContentType,
      submittedAt: new Date()
    });
    
    res.send('Application submitted successfully');
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
};

export const getAllApplications = async (req, res) => {
  try {
    const apps = await Application.findAll({
      attributes: { exclude: ['resumeData'] } // Don't fetch binary blobs in bulk list, optimizes memory load
    });
    res.json(apps);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
};

export const getResume = async (req, res) => {
  try {
    const { id } = req.params;
    const app = await Application.findByPk(id);
    if (!app || !app.resumeData) {
      return res.status(404).send('Resume not found');
    }
    
    res.setHeader('Content-Disposition', `attachment; filename="${app.resumeFileName}"`);
    res.setHeader('Content-Type', app.resumeContentType);
    res.send(app.resumeData);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
};

export const deleteApplication = async (req, res) => {
  try {
    const { id } = req.params;
    const app = await Application.findByPk(id);
    if (!app) {
      return res.status(404).json({ error: 'Application not found' });
    }
    await app.destroy();
    res.send('Application deleted');
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
};

File: Klanvision_backend/src/main/java/com/klanvision/backend/controller/ApplicationController.java
package com.klanvision.backend.controller;

import com.klanvision.backend.dto.ApplicationDTO;
import com.klanvision.backend.model.Application;
import com.klanvision.backend.service.ApplicationService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.List;

@RestController
@RequestMapping("/api/applications")
@CrossOrigin(origins = "*")
public class ApplicationController {

    @Autowired
    private ApplicationService applicationService;

    @PostMapping(consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
    public ResponseEntity<?> submitApplication(
            @RequestParam("jobTitle") String jobTitle,
            @RequestParam("name") String name,
            @RequestParam("dob") String dob,
            @RequestParam("email") String email,
            @RequestParam("phone") String phone,
            @RequestParam("gender") String gender,
            @RequestParam("qualification") String qualification,
            @RequestParam("experience") String experience,
            @RequestParam("skills") String skills,
            @RequestParam(value = "linkedin", required = false) String linkedin,
            @RequestParam(value = "portfolio", required = false) String portfolio,
            @RequestParam("resume") MultipartFile resume) {

        try {
            ApplicationDTO dto = new ApplicationDTO();
            dto.setJobTitle(jobTitle);
            dto.setName(name);
            dto.setDob(dob);
            dto.setEmail(email);
            dto.setPhone(phone);
            dto.setGender(gender);
            dto.setQualification(qualification);
            dto.setExperience(experience);
            dto.setSkills(skills);
            dto.setLinkedin(linkedin);
            dto.setPortfolio(portfolio);

            applicationService.submitApplication(dto, resume);
            return ResponseEntity.ok("Application submitted successfully");
        } catch (IOException e) {
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Error processing resume");
        }
    }

    @GetMapping
    public List<ApplicationDTO> getAllApplications() {
        return applicationService.getAllApplications();
    }

    @GetMapping("/resume/{id}")
    public ResponseEntity<byte[]> getResume(@PathVariable Long id) {
        Application app = applicationService.getApplicationEntity(id);
        return ResponseEntity.ok()
                .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=\"" + app.getResumeFileName() + "\"")
                .contentType(MediaType.parseMediaType(app.getResumeContentType()))
                .body(app.getResumeData());
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<?> deleteApplication(@PathVariable Long id) {
        applicationService.deleteApplication(id);
        return ResponseEntity.ok("Application deleted");
    }
}
